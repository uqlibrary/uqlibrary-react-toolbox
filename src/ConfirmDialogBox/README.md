# ConfirmDialogBox

Pre formatted dialog box for confirmation

## Props

- locale: PropTypes.object
- onAction: PropTypes.func
- onRef: PropTypes.func

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {ConfirmDialogBox} from 'uqlibrary-react-toolbox';
 
const App = () => (
  <div>
    <ConfirmDialogBox onAction={this.deleteItem} onRef={this.setRef} />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
