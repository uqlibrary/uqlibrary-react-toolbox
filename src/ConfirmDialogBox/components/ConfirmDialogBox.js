import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import RaisedButton from 'material-ui/RaisedButton';

export default class ConfirmDialogBox extends Component {

    static propTypes = {
        locale: PropTypes.object,
        onAction: PropTypes.func,
        onCancelAction: PropTypes.func,
        hideCancelButton: PropTypes.bool,
        onRef: PropTypes.func
    };

    static defaultProps = {
        hideCancelButton: false,
        locale: {
            confirmationTitle: 'Confirmation',
            confirmationMessage: 'Are you sure?',
            cancelButtonLabel: 'No',
            confirmButtonLabel: 'Yes'
        }
    };

    constructor(props) {
        super(props);

        this._onCancelAction = this._onCancelAction.bind(this);
        this._hideConfirmation = this._hideConfirmation.bind(this);
        this._onAction = this._onAction.bind(this);

        this.state = {
            isDialogOpen: false
        };
    }

    componentDidMount() {
        this.props.onRef(this);
    }

    componentWillUnmount() {
        this.props.onRef(undefined);
    }

    showConfirmation() {
        this.setState({
            isDialogOpen: true
        });
    }

    _hideConfirmation() {
        this.setState({
            isDialogOpen: false
        });
    }

    _onAction() {
        this._hideConfirmation();
        this.props.onAction();
    }

    _onCancelAction() {
        this._hideConfirmation();
        if (this.props.onCancelAction) {
            this.props.onCancelAction();
        }
    }

    render() {
        const actions = [
            <div className="columns dialog-actions ConfirmDialogBox-actions">
                <div className="column is-hidden-mobile"/>
                {
                    !this.props.hideCancelButton &&
                    <div className="column is-narrow">
                        <RaisedButton label={this.props.locale.cancelButtonLabel}
                                      fullWidth
                                      className='ConfirmDialogBox-actions-cancel'
                                      onTouchTap={this._onCancelAction}/>
                    </div>
                }
                <div className="column is-narrow">
                    <RaisedButton label={this.props.locale.confirmButtonLabel}
                                  fullWidth
                                  className='ConfirmDialogBox-actions-confirm'
                                  secondary
                                  keyboardFocused
                                  onTouchTap={this._onAction}/>
                </div>
                <div className="is-clearfix"/>
            </div>
        ];

        return (
            <Dialog
                title={this.props.locale.confirmationTitle}
                actions={actions}
                modal
                open={this.state.isDialogOpen}>
                {this.props.locale.confirmationMessage}
            </Dialog>
        );
    }
}
