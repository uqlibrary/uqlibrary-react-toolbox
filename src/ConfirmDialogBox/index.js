export {default as ConfirmDialogBox} from './components/ConfirmDialogBox';
