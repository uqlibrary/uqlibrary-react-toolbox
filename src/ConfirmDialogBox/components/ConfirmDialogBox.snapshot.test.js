jest.dontMock('./ConfirmDialogBox');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ConfirmDialogBox from './ConfirmDialogBox';

function setup({locale, hideCancelButton, cancelAction, action}) {
    const props = {
        locale: locale || {
            confirmationTitle: 'Confirmation',
            confirmationMessage: 'Are you sure?',
            cancelButtonLabel: 'No',
            confirmButtonLabel: 'Yes'
        },
        onAction: action || jest.fn(),
        onCancelAction: cancelAction || jest.fn(),
        hideCancelButton: hideCancelButton || false,
        onRef: jest.fn()
    };

    return shallow(<ConfirmDialogBox {...props} />);
}

describe('ConfirmDialogBox snapshots tests', () => {
    it('renders component with yes/no buttons', () => {
        const wrapper = setup({});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders component with yes', () => {
        const wrapper = setup({hideCancelButton: true});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
    it('renders component with customised locale', () => {
        const wrapper = setup({locale: {
            confirmationTitle: 'ENG: Confirmation',
            confirmationMessage: 'ENG: Are you sure?',
            cancelButtonLabel: 'ENG: No',
            confirmButtonLabel: 'ENG: Yes'
        }});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
