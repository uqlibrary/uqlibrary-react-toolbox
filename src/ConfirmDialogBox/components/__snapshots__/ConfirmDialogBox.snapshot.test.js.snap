// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogBox snapshots tests renders component with customised locale 1`] = `
<Dialog
  actions={
    Array [
      <div
        className="columns dialog-actions ConfirmDialogBox-actions"
    >
        <div
            className="column is-hidden-mobile"
        />
        <div
            className="column is-narrow"
        >
            <RaisedButton
                className="ConfirmDialogBox-actions-cancel"
                disabled={false}
                fullWidth={true}
                label="ENG: No"
                labelPosition="after"
                onTouchTap={[Function]}
                primary={false}
                secondary={false}
            />
        </div>
        <div
            className="column is-narrow"
        >
            <RaisedButton
                className="ConfirmDialogBox-actions-confirm"
                disabled={false}
                fullWidth={true}
                keyboardFocused={true}
                label="ENG: Yes"
                labelPosition="after"
                onTouchTap={[Function]}
                primary={false}
                secondary={true}
            />
        </div>
        <div
            className="is-clearfix"
        />
    </div>,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={false}
  repositionOnUpdate={true}
  title="ENG: Confirmation"
>
  ENG: Are you sure?
</Dialog>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes 1`] = `
<Dialog
  actions={
    Array [
      <div
        className="columns dialog-actions ConfirmDialogBox-actions"
    >
        <div
            className="column is-hidden-mobile"
        />
        <div
            className="column is-narrow"
        >
            <RaisedButton
                className="ConfirmDialogBox-actions-confirm"
                disabled={false}
                fullWidth={true}
                keyboardFocused={true}
                label="Yes"
                labelPosition="after"
                onTouchTap={[Function]}
                primary={false}
                secondary={true}
            />
        </div>
        <div
            className="is-clearfix"
        />
    </div>,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={false}
  repositionOnUpdate={true}
  title="Confirmation"
>
  Are you sure?
</Dialog>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes/no buttons 1`] = `
<Dialog
  actions={
    Array [
      <div
        className="columns dialog-actions ConfirmDialogBox-actions"
    >
        <div
            className="column is-hidden-mobile"
        />
        <div
            className="column is-narrow"
        >
            <RaisedButton
                className="ConfirmDialogBox-actions-cancel"
                disabled={false}
                fullWidth={true}
                label="No"
                labelPosition="after"
                onTouchTap={[Function]}
                primary={false}
                secondary={false}
            />
        </div>
        <div
            className="column is-narrow"
        >
            <RaisedButton
                className="ConfirmDialogBox-actions-confirm"
                disabled={false}
                fullWidth={true}
                keyboardFocused={true}
                label="Yes"
                labelPosition="after"
                onTouchTap={[Function]}
                primary={false}
                secondary={true}
            />
        </div>
        <div
            className="is-clearfix"
        />
    </div>,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={false}
  repositionOnUpdate={true}
  title="Confirmation"
>
  Are you sure?
</Dialog>
`;
